crates/router/tests/connectors/worldline.rs

```diff
@@ -0,0 +1,276 @@
+use api_models::payments::{Address, AddressDetails};
+use masking::Secret;
+use router::types::{self, storage::enums, PaymentAddress};
+
+use crate::{
+    connector_auth::ConnectorAuthentication,
+    utils::{self, ConnectorActions, PaymentInfo},
+};
+
+struct Worldline;
+
+impl ConnectorActions for Worldline {}
+impl utils::Connector for Worldline {
+    fn get_data(&self) -> types::api::ConnectorData {
+        use router::connector::Worldline;
+        types::api::ConnectorData {
+            connector: Box::new(&Worldline),
+            connector_name: types::Connector::Worldline,
+            get_token: types::api::GetToken::Connector,
+        }
+    }
+
+    fn get_auth_token(&self) -> types::ConnectorAuthType {
+        types::ConnectorAuthType::from(
+            ConnectorAuthentication::new()
+                .worldline
+                .expect("Missing connector authentication configuration"),
+        )
+    }
+
+    fn get_name(&self) -> String {
+        "worldline".to_string()
+    }
+}
```

Comment:
**Suggestion**: Not necessary to use the `use` statement to import the Worldline struct since it's already in scope within the impl block and the Box::new(&Worldline) creates a new boxed instance of the Worldline struct, but it seems that you want to pass the self instance instead. We can change it to Box::new(self). In the `get_auth_token method`, you can use the `unwrap_or_default` method instead of `expect` to handle the case where the worldline field is None, will return a default value instead of panicking. Also, in the `get_name method`, you can use the `String::from` function instead of `to_string` to create a new string instance.

```suggestion
impl utils::Connector for Worldline {
    fn get_data(&self) -> types::api::ConnectorData {
        types::api::ConnectorData {
            connector: Box::new(self),
            connector_name: types::Connector::Worldline,
            get_token: types::api::GetToken::Connector,
        }
    }

    fn get_auth_token(&self) -> types::ConnectorAuthType {
        types::ConnectorAuthType::from(
            ConnectorAuthentication::new()
                .worldline
                .unwrap_or_default(),
        )
    }

    fn get_name(&self) -> String {
        String::from("worldline")
    }
}
```

Reply:
First of all, @malwarebo, we appreciate your time and effort in reviewing our code. It makes us happy to see members of the community being involved in the development process, specifically the PR reviews!

> Also, in the `get_name method`, you can use the `String::from` function instead of `to_string` to create a new string instance.

Is there any specific reason why you're suggesting `String::from()`instead of `to_string()`? As far as I understand, there is a heap allocation involved in both cases, and both versions should have similar performance, if not the same, at least in recent Rust versions. Let me know if I'm missing something.

Thanks again for being involved!

Reply:
Happy to contribute @SanchithHegde . 
You are right, the performance difference between these two functions is not significant unless you are dealing with large scale data and I am not aware about that in your case. So I suggested the use of `String::from`. Also, it is more versatile in creating a new String from different types like slices, C-style and other String objects plus it is more readable. You can totally make a best call here. :)

Reply:
@malwarebo Thanks for your insights. This is a test file and the the struct name for both test and actual code is same, that caused the confusion here. I renamed the test struct to "WorldlineTest".